cpp/include/cudf/strings/split/partition.hpp

```diff
@@ -24,6 +24,8 @@ namespace strings {
 /**
  * @addtogroup strings_split
  * @{
+ * @file strings/split/partition.hpp
```

Comment:
why is the file name here, and not in other files?

Reply:
The name is optional but if there are two files with the same name in different directories (we have a few) then specifying the name can help distinguish them. I found that doxygen is ignoring my suggestions but I left some of these in anyway.